File: 02_agent_patterns/01_agent_loop_basics/agent-loop-target/agent_loop_target/weather.py
"""Bug 4 (incorrect API response parsing): pull fields out of a weather payload."""


def parse_current_temperature(response):
    """Return the current temperature in Celsius from a weather API response.

    The API returns a nested JSON object shaped like::

        {
            "location": {"name": "San Francisco", "region": "CA"},
            "current": {
                "temperature_c": 21.5,
                "humidity": 60,
                "condition": "Partly cloudy"
            }
        }

    Args:
        response: the decoded JSON response (a dict) with the structure above.

    Returns:
        The current temperature in Celsius as a number.
    """
    return response["temperature_c"]


def parse_condition(response):
    """Return the human-readable ``condition`` string from the same response.

    See :func:`parse_current_temperature` for the response structure.
    """
    return response["current"]["condition"]

File: 02_agent_patterns/01_agent_loop_basics/agent-loop-target/go-example/weather.go
package agentloop

// ParseCurrentTemperature returns the current temperature in Celsius from a
// decoded weather API response shaped like:
//
//	map[string]any{
//	    "location": map[string]any{"name": "San Francisco", "region": "CA"},
//	    "current": map[string]any{
//	        "temperature_c": 21.5,
//	        "humidity":      60,
//	        "condition":     "Partly cloudy",
//	    },
//	}
func ParseCurrentTemperature(resp map[string]any) float64 {
	return resp["temperature_c"].(float64)
}

// ParseCondition returns the human-readable condition string from the same
// response. It already works — use it to cross-reference the correct nesting.
func ParseCondition(resp map[string]any) string {
	current := resp["current"].(map[string]any)
	return current["condition"].(string)
}
